artifact/artifact.sh

```diff
@@ -1,48 +1,89 @@
+        elif [ "$CI_PROVIDER" = "circle" ]; then
+            REPO_SLUG="${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}"
+            COMMIT=$CIRCLE_SHA1
+            TAG=${CIRCLE_TAG:-}
+            BRANCH=${CIRCLE_BRANCH:-}
+            PR_BRANCH=${CIRCLE_PR_NUMBER:-}
+        fi
+
+        if [ -z "$PR_BRANCH" ]; then
+            IS_PULL_REQUEST=false
+        else
+            IS_PULL_REQUEST=true
+        fi
+
+        DOCKER_REPO="tidepool/${REPO_SLUG#*/}"
+
+        echo "${DOCKER_PASSWORD}" | docker login --username "${DOCKER_USERNAME}" --password-stdin
+        if [ "${REPO_SLUG:-}" == "tidepool-org/blip" ] || [ "${REPO_SLUG:-}" == "tidepool-org/uploader" ]; then
+            # Build blip image
             RX_ENABLED=${RX_ENABLED-false}
-            if [[ ",${RX_ENABLED_BRANCHES:-}," = *",${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH},"* ]]; then RX_ENABLED=true; fi
-            DOCKER_BUILDKIT=1 docker build --tag "${DOCKER_REPO}" --build-arg ROLLBAR_POST_SERVER_TOKEN="${ROLLBAR_POST_SERVER_TOKEN:-}" --build-arg LAUNCHDARKLY_CLIENT_TOKEN="${LAUNCHDARKLY_CLIENT_TOKEN:-}" --build-arg REACT_APP_GAID="${REACT_APP_GAID:-}" --build-arg RX_ENABLED="${RX_ENABLED:-}" --build-arg TRAVIS_COMMIT="${TRAVIS_COMMIT:-}" .
+            if [[ ",${RX_ENABLED_BRANCHES:-}," = *",${PR_BRANCH:-$BRANCH},"* ]]; then RX_ENABLED=true; fi
+            DOCKER_BUILDKIT=1 docker build --tag "${DOCKER_REPO}" --build-arg ROLLBAR_POST_SERVER_TOKEN="${ROLLBAR_POST_SERVER_TOKEN:-}" --build-arg LAUNCHDARKLY_CLIENT_TOKEN="${LAUNCHDARKLY_CLIENT_TOKEN:-}" --build-arg REACT_APP_GAID="${REACT_APP_GAID:-}" --build-arg RX_ENABLED="${RX_ENABLED:-}" --build-arg TRAVIS_COMMIT="${COMMIT:-}" .
         else
+            # Build other images
             docker build --tag "${DOCKER_REPO}" .
         fi
 
-        if [ "${TRAVIS_BRANCH:-}" == "master" ] && [ "${TRAVIS_PULL_REQUEST_BRANCH:-}" == "" ]; then
+        if [ "${BRANCH:-}" == "master" ] && [ $IS_PULL_REQUEST == false ]; then
+            # Push master branch image
             docker push "${DOCKER_REPO}"
         fi
-        if [ -n "${TRAVIS_TAG:-}" ]; then
-            docker tag "${DOCKER_REPO}" "${DOCKER_REPO}:${TRAVIS_TAG}"
-            docker push "${DOCKER_REPO}:${TRAVIS_TAG}"
+
+        if [ -n "${TAG:-}" ]; then
+            # Push git tag image
+            docker tag "${DOCKER_REPO}" "${DOCKER_REPO}:${TAG}"
+            docker push "${DOCKER_REPO}:${TAG}"
         fi
-        if [ -n "${TRAVIS_BRANCH:-}" ] && [ -n "${TRAVIS_COMMIT:-}"  ]; then
-            if [ -n "${TRAVIS_PULL_REQUEST_BRANCH}" ]
+
+        if [ -n "$BRANCH" ] && [ -n "$COMMIT" ]; then
+            if [ $IS_PULL_REQUEST == true && "$CI_PROVIDER" = "travis" ]
             then
-                BRANCH=$(echo -n ${TRAVIS_PULL_REQUEST_BRANCH} | tr / -)
+                BRANCH=$(echo -n ${PR_BRANCH} | tr / -)
             else
-                BRANCH=$(echo -n ${TRAVIS_BRANCH} | tr / -)
+                BRANCH=$(echo -n ${BRANCH} | tr / -)
```

Comment:
Ah, now I see how using the PR number as the `PR_BRANCH` is resolved.

Reply:
However, this still results in two images with use the same tags (one from feature branch build and one from PR build). I think we'll want to change this. See the above mentioned Slack thread.